crates/api_crud/src/site/read.rs

```diff
@@ -99,8 +109,25 @@ pub async fn get_site(
       discussion_languages,
     })
   } else {
+    // filter oauth_providers for public access
+    filter_oauth_providers(&local_user_view, &mut site_response);
     None
   };
 
   Ok(Json(site_response))
 }
+
+fn filter_oauth_providers(
+  local_user_view: &Option<LocalUserView>,
+  site_response: &mut GetSiteResponse,
+) {
+  if local_user_view.is_none()
+    || !local_user_view
+      .clone()
+      .expect("user_view must be set")
+      .local_user
+      .admin
```

Comment:
this looks like it can be done cleaner with smth like !local_user_view.map(|user| user.local_user.admin).unwrap_or(false)

Reply:
`is_some_and` method would be more readable